process/sync/baseSync.go

```diff
@@ -106,21 +119,32 @@ func (boot *baseBootstrap) loadBlocks(
 	}
 
 	if currentNonce <= blockFinality {
-		return process.ErrNotEnoughValidBlocksInStorage
+		err = process.ErrNotEnoughValidBlocksInStorage
+		lastNotarized = make(map[uint32]uint64, 0)
+		finalNotarized = make(map[uint32]uint64, 0)
+		currentNonce = 0
```

Comment:
call a cleanup function here instead of checking err == nil.

Reply:
Refactored